p2p/src/sync/mock_consensus.rs

```diff
@@ -0,0 +1,869 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+#![allow(clippy::new_without_default, clippy::result_unit_err)]
+use common::{
+    chain::{
+        block::{consensus_data::ConsensusData, Block, BlockHeader},
+        transaction::Transaction,
+    },
+    primitives::{id, Id, Idable},
+};
+use parity_scale_codec::{Decode, Encode};
+use rand::Rng;
+use std::collections::BTreeMap;
+use tokio::sync::{mpsc, oneshot};
+
+pub type Hash = u64;
+pub type Amount = u64;
+
+trait IdForBlockHeader {
+    fn get_hdr_id(&self) -> Id<Block>;
+}
+
+impl IdForBlockHeader for BlockHeader {
+    fn get_hdr_id(&self) -> Id<Block> {
+        Id::new(&id::hash_encoded(self))
+    }
+}
```

Comment:
I've been thinking to move the get_id to the header for a while. I think it's time for that since most PRs are closed. I'll think of something now and fix this.

Btw, while I understand why you did this:`Id::new(&id::hash_encoded(self))`, this is the stuff of nightmares. We should never repeat implementations manually.